cwltool/provenance.py

```diff
@@ -138,14 +134,23 @@ def _local_path(posix_path):
 
 
 def _whoami():
-    # type: () -> Tuple[str,str]
+    # type: () -> Tuple[Text,Text]
     """Return the current operating system account as (username, fullname)."""
     username = getuser()
-    fullname = username
-    if GET_PW_NAM:
-        pwnam = getpwnam(username)
-        if pwnam:
-            fullname = pwnam.pw_gecos.split(",", 1)[0]
+    try:
+        if onWindows():
+            get_user_name = ctypes.windll.secur32.GetUserNameExW  # type: ignore
```

Comment:
:heart_eyes:

Reply:
Had to grab a Windows machine to make sure it works properly ;)